File: app/src/Song.ts
import * as _ from 'lodash';

class Song {
    id: number;
    album: string;
    artist: string;
    genre: string;
    mime: string;
    name: string;
    stream_location: string;
    year: number;
    duration: number;
    cover: string;

    constructor(data: any) {
        if (data) {
            _.assign(this, data);
        }
    }

    getCoverUrl(): string {
        return this.cover;
    }


}

export default Song;

File: models/Song.go
package models

import (
	"github.com/jinzhu/gorm"
)

// Song model.
type Song struct {
	gorm.Model

	Name string `gorm:"not null"`

	Artist   *Artist `gorm:"ForeignKey:ArtistID"`
	ArtistID NullInt64

	Album   *Album `gorm:"ForeignKey:AlbumID"`
	AlbumID NullInt64

	Cover   *Image `gorm:"ForeignKey:CoverID"`
	CoverID NullInt64

	Year NullInt64

	Genre    NullString
	Duration NullFloat64
	Mime     string `gorm:"not null"`
	Path     string `gorm:"not null"`
	Played   uint   `gorm:"not null"`
}
